scripts/Diffraction/isis_powder/routines/common.py

```diff
@@ -109,7 +116,7 @@ def generate_run_numbers(run_number_string):
     :return: A list of run numbers generated from the string
     """
     # Check its not a single run
-    if isinstance(run_number_string, int) or run_number_string.isdigit():
+    if isinstance(run_number_string, int):
```

Comment:
This does not make sense anymore. You're checking it's an `int` then casting it to an `int`. So I'm guess this fails when it's a `str`.

Is this function still expected to take strings? If so I think we still need another case here? 
1) is string run number
2) is int run number
3) is a list of run numbers

If not then we should update the docstring, which also needs a `:param:` entry anyway and the variable name.